pennylane/labs/resource_estimation/templates/trotter.py

```diff
@@ -0,0 +1,221 @@
+            "n": n,
+            "order": order,
+            "first_order_expansion": first_order_expansion,
+        }
+
+    @classmethod
+    def resource_rep(cls, n, order, first_order_expansion) -> CompressedResourceOp:
+        params = {
+            "n": n,
+            "order": order,
+            "first_order_expansion": first_order_expansion,
+        }
+        return CompressedResourceOp(cls, params)
+
+    @classmethod
+    def resources(cls, *args, **kwargs) -> Dict[CompressedResourceOp, int]:
+        """Returns a dictionary containing the counts of each operator type used to
+        compute the resources of the operator."""
+        return cls._resource_decomp(*args, **kwargs)
+
+
+class ResourceTrotterizedQfunc(TrotterizedQfunc, ResourceOperator):
+    """An internal class which facilitates :code:`qml.resource_trotterize`."""
+
+    @staticmethod
+    def _resource_decomp(
+        n, order, qfunc_compressed_reps, **kwargs
+    ) -> Dict[CompressedResourceOp, int]:
+        k = order // 2
+        gate_types = defaultdict(int, {})
+
+        if order == 1:
+            for cp_rep in qfunc_compressed_reps:
+                gate_types[cp_rep] += n
+            return gate_types
+
+        for cp_rep in qfunc_compressed_reps:
+            gate_types[cp_rep] += 2 * n * (5 ** (k - 1))
+        return gate_types
+
+    def resource_params(self) -> dict:
+        with qml.QueuingManager.stop_recording():
+            with qml.queuing.AnnotatedQueue() as q:
+                base_hyper_params = ("n", "order", "qfunc", "reverse")
+
+                qfunc_args = self.parameters
+                qfunc_kwargs = {
+                    k: v for k, v in self.hyperparameters.items() if not k in base_hyper_params
+                }
+
+                qfunc = self.hyperparameters["qfunc"]
+                qfunc(*qfunc_args, wires=self.wires, **qfunc_kwargs)
+
+        try:
+            qfunc_compressed_reps = tuple(op.resource_rep_from_op() for op in q.queue)
+
+        except AttributeError as error:
+            raise ResourcesNotDefined(
+                "Every operation in the TrotterizedQfunc should be a ResourceOperator"
+            ) from error
+
+        return {
+            "n": self.hyperparameters["n"],
+            "order": self.hyperparameters["order"],
+            "qfunc_compressed_reps": qfunc_compressed_reps,
+        }
+
+    @classmethod
+    def resource_rep(cls, n, order, qfunc_compressed_reps, name=None) -> CompressedResourceOp:
+        params = {
+            "n": n,
+            "order": order,
+            "qfunc_compressed_reps": qfunc_compressed_reps,
+        }
+        return CompressedResourceOp(cls, params, name=name)
+
+    def resource_rep_from_op(self) -> CompressedResourceOp:
+        """Returns a compressed representation directly from the operator"""
+        return self.__class__.resource_rep(**self.resource_params(), name=self._name)
+
+
+def resource_trotterize(qfunc, n=1, order=2, reverse=False):
```

Comment:
Is `qml.trotterize` the only function that has a resource counterpart function?

Reply:
Yes, but it will be deleted once we move the functionality to Core-PL since the only reason we needed it was to ensure we were passing `ResourceTrotterizedQfunc` instead of just `TrotterizedQfunc`